src/common/libioencode/ioencode.c

```diff
@@ -20,9 +20,33 @@
 #include <jansson.h>
 
 #include "src/common/libutil/macros.h"
+#include "src/common/libccan/ccan/base64/base64.h"
 
 #include "ioencode.h"
 
+
+static json_t *data_encode_base64 (const char *stream,
+                                   const char *rank,
+                                   const char *data,
+                                   int len)
+{
+    ssize_t n;
+    json_t *o = NULL;
+    char *dest = NULL;
+    size_t destlen = base64_encoded_length (len);
+
+    if ((dest = calloc (1, destlen))
```

Comment:
can call `malloc` instead of `calloc`?  Since all data will be overwritten in the `base64_encode()` below.

Also on the decode path below in two places.

Reply:
This is true for `base64_encoded_length()`, but `base64_decoded_length()` returns the _minimum_ buffer size for decode, so some bytes would otherwise be uninitialized and it seemed safer to set those to zero.

I can change the calloc() here if you feel that is important.

Reply:
Actually I just realized this comment from `base64_encoded_length()`:
```c
/**
 * base64_encoded_length - Calculate encode buffer length
 * @param srclen the size of the data to be encoded
 * @note add 1 to this to get null-termination
 * @return Buffer length required for encode
 */
```

So this code needs to be fixed to add 1 to destlen. Then, to ensure NUL termination, it is probably safest to use `calloc` here just in case `base64_encode()` doesn't write the NUL.

Reply:
> I can change the calloc() here if you feel that is important. 

Nah, if it does need to an initialize, then its fine.  I thought the entire buffer would be overwritten right after.  I think one decode path does a `calloc` then `memcpy`, so that could probably be a `malloc` instead.

Reply:
Ah, you're right. That was the existing code (and the common code path), so I should probably change that back to a `malloc()`. It was changed inadvertently when I refactored some code.

Reply:
Ok, I forced a push changing the useless use of `calloc()` back to `malloc()`. Thanks!